project/AkkaBuild.scala

```diff
@@ -404,17 +404,26 @@ object AkkaBuild extends Build {
     mavenLocalResolverSettings ++
     JUnitFileReporting.settings ++ StatsDMetrics.settings
 
-  def akkaPreviousArtifact(id: String): Def.Initialize[Option[sbt.ModuleID]] = Def.setting {
+  def akkaPreviousArtifacts(id: String): Def.Initialize[Set[sbt.ModuleID]] = Def.setting {
     if (enableMiMa) {
-      val version: String = "2.4.0" // FIXME verify all 2.3.x versions
-      val fullId = crossVersion.value match {
-        case _ : CrossVersion.Binary => id + "_" + scalaBinaryVersion.value
-        case _ : CrossVersion.Full => id + "_" + scalaVersion.value
-        case CrossVersion.Disabled => id
+      val versions = {
+        val akka23Versions = Seq("2.3.11", "2.3.12", "2.3.13", "2.3.14")
+        val akka24Versions = Seq("2.4.0")
+        scalaBinaryVersion.value match {
+          case "2.11" if !Seq(
+            "akka-cluster-sharding",
+            "akka-cluster-tools",
+            "akka-persistence",
```

Comment:
can we make this a bit more named methods? I also was a bit confused what actually will be tested